templates/hooks/certificate/sdk_read_one_post_set_output.go.tpl

```diff
@@ -1,7 +1,11 @@
-    err = rm.writeCertificateToSecret(ctx, *resp.Certificate, r.ko.ObjectMeta)
-    if err != nil && strings.HasPrefix(err.Error(), "RequestInProgressException") {
-        return &resource{ko}, ackrequeue.NeededAfter(err, ackrequeue.DefaultRequeueAfterDuration)
+  err = rm.writeCertificateToSecret(ctx, *resp.Certificate, r.ko.ObjectMeta)
+  if err != nil {
+    // If the secret cannot be written to and the certificate is marked for deletion
+    // mark the object as NotFound, this aids in gracefully deleting certificates where the
+    // specified secret is deleted first
+    if r.ko.ObjectMeta.DeletionTimestamp != nil {
+      return nil, ackerr.NotFound
```

Comment:
@a-hilaly does returning a NotFound error, trigger creation here?

Reply:
Alternatively we could just return the object as is if the certificate is marked for deletion and the certificate doesn't exists

```
  if err != nil {
    // If the secret cannot be written to and the certificate is marked for deletion
    // return the object as is with no error, this aids in gracefully deleting certificates where the
    // specified secret is deleted first
    if r.ko.ObjectMeta.DeletionTimestamp != nil {
      return &resource{ko}, nil
    }
   return nil, err
}
```

Reply:
How about only attempting to write to a secret if the resource isn't marked for deletion?

Reply:
> @a-hilaly does returning a NotFound error, trigger creation here?

Correct, we shouldn't return a NotFound here

Reply:
keep in mind that the reconciliation logic enters a different path when a resource is marked for deletion: https://github.com/aws-controllers-k8s/runtime/blob/main/pkg/runtime/reconciler.go#L287-L302

Reply:
I like this approach, will do that instead.